oeps/oep-0036-microservice-routing-layer.rst

```diff
@@ -0,0 +1,99 @@
+===================================
+OEP-36: Microservice Routing Layer
+===================================
+
+.. list-table::
+
+  * - OEP
+    - :doc:`oep-0036-microservice-routing-layer`
+  * - Title
+    - Microservice Routing Layer
+  * - Last-Modified
+    - 2019-04-25
+  * - Author
+    - Fred Smith <derf@edx.org>
+  * - Arbiter
+    - David Joy <djoy@edx.org>
+  * - Status
+    - Provisional
+  * - Type
+    - Architecture
+  * - Created
+    - 2019-04-25
+
+Abstract
+========
+
+Proposes a system with which users can interact with microservice applications
+using a layer 7 routing framework. 
+
+Motivation
+==========
+
+Microservices are currently implemented using layer 4 (DNS) based routing. Each
+microservice occupies a unique hostname, with dedicated ELB or CDN 
+infrastructure to support routing user traffic to the service. This pattern
+is simple to monitor, deploy, and manage, but slow and expensive to build,
+and regularly causes problems with cross service user information sharing. In
```

Comment:
Also, maybe add some details to the cross-service issues?  CORS, inability to share localStorage, other?

On the flip side, I don't know if this affects cookies in the following ways:
* Did we have large subdomain specific cookies, that now all will be sent to every request across the entire site?  Also, could we more easily hit our cookie size limit of 4kb?
* Could we have any cookie name conflicts that were previously namespaced to the subdomain?